specs/dwritecore/DWriteCore.md

```diff
@@ -259,5 +1127,87 @@ DWRITE_BEGIN_INTERFACE(IDWriteBitmapRenderTarget2, "C553A742-FC01-44DA-A66E-B8B9
     /// </summary>
     STDMETHOD(GetBitmapData)(_Out_ DWRITE_BITMAP_DATA_BGRA32* bitmapData) PURE;
 };
-#endif // DWRITE_CORE
+```
+
+## IDWriteFontSet4 Interface
+
+```c++
+DWRITE_BEGIN_INTERFACE(IDWriteFontSet4, "EEC175FC-BEA9-4C86-8B53-CCBDD7DF0C82") : IDWriteFontSet3
+{
+    /// <summary>
+    /// Computes derived font axis values from the specified font weight, stretch, style, and size.
+    /// </summary>
+    /// <param name="inputAxisValues">Pointer to an optional array of input axis values. Any axes present
+    /// in this array are not included in the output. This is so explicit axis values take precedence over 
+    /// derived axis values.</param>
+    /// <param name="inputAxisCount">Size of the array of input axis values.</param>
+    /// <param name="fontWeight">Font weight, used to compute "wght" axis value.</param>
+    /// <param name="fontStretch">Font stretch, used to compute "wdth" axis value.</param>
+    /// <param name="fontStyle">Font style, used to compute "slnt" and "ital" axis values.</param>
+    /// <param name="fontSize">Font size in DIPs, used to compute "opsz" axis value. If this parameter is zero,
+    /// no "opsz" axis value is added to the output array.</param>
+    /// <param name="outputAxisValues">Pointer to an output array to which derived axis values are written.
+    /// The size of this array must be at least DWRITE_STANDARD_FONT_AXIS_COUNT (5). The return value is 
+    /// the actual number of axis values written to this array.</param>
+    /// <returns>Returns the actual number of derived axis values written to the output array.</returns>
+    STDMETHOD_(UINT32, GetDerivedFontAxisValues)(
+        _In_reads_opt_(inputAxisCount) DWRITE_FONT_AXIS_VALUE const* inputAxisValues,
+        UINT32 inputAxisCount,
+        DWRITE_FONT_WEIGHT fontWeight,
+        DWRITE_FONT_STRETCH fontStretch,
+        DWRITE_FONT_STYLE fontStyle,
+        float fontSize,
+        _Out_writes_to_(DWRITE_STANDARD_FONT_AXIS_COUNT, return) DWRITE_FONT_AXIS_VALUE* outputAxisValues
```

Comment:
Is this `DWRITE_STANDARD_FONT_AXIS_COUNT` already defined somewhere?

Reply:
No, that is a new #define.
```
#define DWRITE_STANDARD_FONT_AXIS_COUNT 5
```
Should I give that its own heading?

Reply:
DISCUSSION: Why only up to five total output values?  There are only five well-known defined output axes - weight, width, slant, italics, optical-size.  Those are the ones that can be derived from the input values passed in.

What if there's a sixth default axis?  We'd add a GetDerivedFontAxisValues2 that takes new values to do the derivation. This is for WWS->typographic migration.

CONSIDER: Should the name of this method be more "it's a bridge" like "DeriveAxisValuesFromWWSModel" or similar, or "ConvertWeightWidthStyleToFontAxis"

Reply:
After discussing with various people, I'm going with `DeriveAxisValuesFromWeightStretchStyle`.

Reply:
After further discussion, I renamed the method to ConvertWeightStretchStyleToFontAxisValues.
"FontAxisValues" instead of "AxisValues" for consistency with other DWrite methods.
"Convert" instead of "Derive" because it's more common and is already used in other DWrite methods.